[docs] [migrate] Added more known issues to troubleshoot section (#15240)
yugabyte/yugabyte-db: docs/content/preview/migrate/troubleshoot.md

---

### Issue #137: [MYSQL] Spatial datatype migration is not yet supported

**GitHub link**: [Issue #137](https://github.com/yugabyte/yb-voyager/issues/137)

> **Description**: If your MYSQL schema contains spatial datatypes, the migration will not complete as this migration type is not yet supported by YugabyteDB Voyager. This will require extra dependencies such as [PostGIS](https://postgis.net/) to be installed.

Reviewer: This will require extra dependencies - what is 'this'? installed why and where? in the future, for a migration that won't work anyway?

---

### Issue #612: [PostgreSQL] Adding primary key to a partitioned table results in an error

**GitHub link**: [Issue #612](https://github.com/yugabyte/yb-voyager/issues/612)

> **Description**: If you have a partitioned table in which primary key is added later using `ALTER TABLE`, then it fails with the following error:

Reviewer: then it fails -> then the migration fials?

Author: table creation fails . Ill update it.

```output
ERROR: adding primary key to a partitioned table is not yet implemented (SQLSTATE XX000)
```

> **Workaround**: Add primary key in the `CREATE TABLE` statement.

Reviewer: manual intervention needed?

**Example**

An example schema on the source database is as follows:

```sql
CREATE TABLE public.sales_region (
    id integer NOT NULL,
    amount integer,
    branch text,
    region text NOT NULL
)
PARTITION BY LIST (region);

ALTER TABLE ONLY public.sales_region ADD CONSTRAINT sales_region_pkey PRIMARY KEY (id, region);
```

Suggested change to the schema is as follows:

```sql
CREATE TABLE public.sales_region (
    id integer NOT NULL,
    amount integer,
    branch text,
    region text NOT NULL,
    PRIMARY KEY(id, region)
)
PARTITION BY LIST (region);
```

---

### Issue #49: Index on timestamp column should be imported as ASC (Range) index to avoid sequential scans

**GitHub link**: [Issue #49](https://github.com/yugabyte/yb-voyager/issues/49)

> **Description**: If there is an index on a timestamp column, those indexes should be imported as a range index as most queries relying on such columns will be using the range predicates. So sequential scans will be avoided and indexed scans will be accessible.

Reviewer: ```suggestion
**Description**: If there is an index on a timestamp column, the index should be imported as a range index automatically, as most queries relying on timestamp columns use range predicates. This avoids sequential scans and makes indexed scans accessible.
```
